lib/sequencescape_client.rb

```diff
@@ -35,44 +35,43 @@ def self.update_extraction_attributes(instance, attrs, username = 'test')
   end
 
   def self.purpose_by_name(name)
-    client.plate_purpose.all.select { |p| p.name===name }.first
+    client.plate_purpose.all.find { |p| p.name === name }
```

Comment:
This isn't great, as we end up pulling back all our purposes each time. Not quite sure why this doesn't use plate creators like limber et al. Suppose we don't have a user at this point?

A better bet would probably to build a cache of uuids for each purpose name and use that instead.